iot-hub/Samples/device/DeviceReconnectionSample/ColorConsoleLogger.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    public class ColorConsoleLogger : ILogger
+    {
+        private readonly string _name;
+        private readonly ColorConsoleLoggerConfiguration _config;
+
+        public ColorConsoleLogger(string name, ColorConsoleLoggerConfiguration config)
+        {
+            _name = name;
+            _config = config;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel == _config.LogLevel;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            if (_config.EventId == 0 || _config.EventId == eventId.Id)
```

Comment:
What is 0?

Reply:
EventId when set to 0 will log all events

Reply:
My point it is a magic number that an uninitiated wouldn't know. Can we make it a constant somewhere discoverable?

Reply:
Sure, I can add it as a constant and add a note explaining what it signifies.